code/sonalyze/sonalyze.go

```diff
@@ -85,79 +69,163 @@ func commandLine() (Command, string) {
 		fmt.Fprintf(out, "  help     - print this message\n")
 		fmt.Fprintf(out, "Each command accepts -h to further explain options.\n")
 		os.Exit(0)
+
+	case "version":
+		// Must print version on stdout, and the features() thing is required by some tests.
+		// "short" indicates that we're only parsing the first 8 fields (v0.6.0 data).
+		fmt.Printf("sonalyze-go version(%s) features(short_untagged_sonar_data)\n", SonalyzeVersion)
+		os.Exit(0)
+
+	default:
+		anyCmd, verb := stdhandler.ParseVerb(cmdName, maybeVerb)
+		if anyCmd == nil {
+			fmt.Fprintf(out, "Required operation missing, try `sonalyze help`\n")
+			os.Exit(2)
+		}
+
+		fs := flag.NewFlagSet(cmdName, flag.ExitOnError)
+		fs.Usage = func() {
+			restargs := ""
+			if _, ok := anyCmd.(SetRestArgumentsAPI); ok {
+				restargs = " [-- logfile ...]"
+			}
+			fmt.Fprintf(
+				out,
+				"Usage: %s %s [options]%s\n\n",
+				cmdName,
+				maybeVerb,
+				restargs,
+			)
+			for _, s := range anyCmd.Summary() {
+				fmt.Fprintln(out, "  ", s)
+			}
+			fmt.Fprint(out, "\nOptions:\n\n")
+			fs.PrintDefaults()
+			if restargs != "" {
+				fmt.Fprintf(out, "  logfile ...\n    \tInput data files\n")
+			}
+		}
+
+		err := stdhandler.ParseArgs(verb, args, anyCmd, fs)
+		if err != nil {
+			fmt.Fprint(out, err.Error())
+			os.Exit(2)
+		}
+
+		if fhCmd, ok := anyCmd.(FormatHelpAPI); ok {
+			if h := fhCmd.MaybeFormatHelp(); h != nil {
+				PrintFormatHelp(out, h)
+				os.Exit(0)
+			}
+		}
+
+		stop, err := stdhandler.StartCPUProfile(anyCmd.CpuProfileFile())
+		if err != nil {
+			return err
+		}
+		if stop != nil {
+			defer stop()
+		}
+
+		if cmd, ok := anyCmd.(RemotableCommand); ok && cmd.RemotingFlags().Remoting {
+			return remoteOperation(cmd, verb, os.Stdin, os.Stdout, out)
+		}
+
+		// We are running against a local cluster store.
+		//
+		// On return, close all open directories after flushing any pending output, cancel all
+		// pending input and return errors from blocked reading operations.
+		//
+		// Note, we are dependent on nobody calling Exit() after this point.
+		defer sonarlog.Close()
+
+		return stdhandler.HandleCommand(anyCmd, os.Stdin, os.Stdout, out)
+	}
+	panic("Unreachable")
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+//
+// Command line parsing and execution helpers that need to work both for interactive ("actual
+// command line") and non-interactive ("daemon") uses.
+
+type StandardCommandLineHandler struct {
+}
+
+func (h *StandardCommandLineHandler) ParseVerb(cmdName, maybeVerb string) (cmd Command, verb string) {
+	switch maybeVerb {
 	case "add":
 		cmd = new(add.AddCommand)
+	case "daemon":
+		cmd = daemon.New(h)
```

Comment:
FIXME?  Passing `h` here would allow a daemon command to respond to a daemon verb?  Investigate!